pytorch_lightning/loggers/wandb.py

```diff
@@ -159,6 +159,7 @@ def version(self) -> Optional[str]:
         # don't create an experiment if we don't have one
         return self._experiment.id if self._experiment else self._id
 
+    @rank_zero_only
```

Comment:
@awaelchli this is correct, right? All the checkpoints should be uploaded on rank zero only?

Reply:
I think we should move it down to the  `if self._log_model`. 
We found that it is a good practice to let loggers track the internal state on each rank and only prevent IO etc. from happening on rank > 0.

Reply:
But experiment is initiated on rank_zero, so in that case, will it make any difference?

Reply:
@rohitgr7. I think what Adrian meant was:
* Logger are user facing API and they don't necessarily know about what `rank_zero_experiment` truly means. 
* If they create their own custom logger and misread the documentation, it will start to mess up with ddp.

I think it is better for us to handle this internally than to expose this decorator. I hope it makes sense.

Best regards,
T.X

Reply:
I don't think this is what Adrian meant.

Reply:
yes, it's not exactly what I meant. 
I agree that we should use the decorators sparsely and when possible to guard the specific code in the methods with an if statement. Even if we don't want to do logging on other ranks, certain code still needs to be exectuted in order to keep the state of the objects in sync where it matters. One example is  the experiment folder, it ideally should be tracked on all ranks so that files can be accessed even if we don't write logs from all processes. 

This was the motivation behind my comment about the decorator in finalize, to be consistent with this rational, and it is not an issue here, becaue finalize doesn't really do anything special. This comment was more about thinking in the future when finalize might do something else that depends on the rank.

Reply:
yep agreed, but for now, even if we don't add it, it won't make a difference since on other processes `self._experiment` will be `None` so it won't do anything. I suggest keeping this just to be consistent with other loggers for now and we can make the changes in the future if required.